src/nodeViews/createReactNodeViewConstructor.tsx

```diff
@@ -85,38 +82,36 @@ export type ReactNodeViewConstructor = (
  * ProseMirror will render content nodes into this element.
  */
 export function createReactNodeViewConstructor(
-  reactNodeViewConstructor: ReactNodeViewConstructor,
+  ReactComponent: ComponentType<NodeViewComponentProps>,
   registerElement: RegisterElement
 ) {
   function nodeViewConstructor(
     node: Node,
     editorView: EditorView,
     getPos: () => number,
-    decorations: readonly Decoration[],
-    innerDecorations: DecorationSource
+    decorations: readonly Decoration[]
   ): NodeView {
-    const reactNodeView = reactNodeViewConstructor(
-      node,
-      editorView,
-      getPos,
-      decorations,
-      innerDecorations
-    );
-
     let componentRef: NodeViewWrapperRef | null = null;
 
-    const { dom, contentDOM, component: ReactComponent } = reactNodeView;
+    const domElementType = node.isInline ? "span" : "div";
+    const dom = document.createElement(domElementType);
+    dom.dataset["pmnodetype"] = node.type.name;
+    dom.dataset["pmdom"] = "true";
```

Comment:
I added these for two reasons:

1. They make the DOM a little clearer when you inspect it; it's much easier to understand what all these divs and spans are doing when they have these data attributes
2. If someone _really_ needs to for some reason, they can select these with CSS or Javascript. There are still queries that would be very challenging to make (like selecting the `dom` wrapper for a node with a specific attribute value), but I think this at least narrows down those edge cases considerably.